modernpython/langPack.py

```diff
@@ -82,30 +256,151 @@ def set_enum_classes(new_enum_classes):
-                with open(template_path, encoding="utf-8") as f:
-                    args = {
-                        "data": class_details,
-                        "template": f,
-                        "partials_dict": partials,
-                    }
-                    output = chevron.render(**args)
-                file.write(output)
+def set_cim_data_type_classes(new_cim_data_type_classes):
+    for new_class in new_cim_data_type_classes:
+        cim_data_type_classes[new_class] = new_cim_data_type_classes[new_class]
 
+def is_cim_data_type_class(name):
+    if name in cim_data_type_classes:
+        return cim_data_type_classes[name]
+
+def has_unit_attribute(attributes):
+    for attr in attributes:
+        if attr["label"] == "unit":
+            return True
+    return False
+
+def run_template(version_path, class_details):
+    if class_details["class_name"] == 'PositionPoint':
+        #this class is created manually to support types conversions
+        return
+    elif class_details["is_a_primitive"] is True:
+        # Primitives are never used in the in memory representation but only for
+        # the schema
+        run_template_primitive(version_path, class_details, primitive_template_files)
+    elif class_details["is_a_cim_data_type"] is True:
+        # Datatypes based on primitives are never used in the in memory
+        # representation but only for the schema
+        run_template_cimdatatype(version_path, class_details, cimdatatype_template_files)
+    elif class_details["has_instances"] is True:
+        run_template_enum(version_path, class_details, enum_template_files)
+    else:
+        run_template_schema(version_path, class_details, template_files)
 
+def run_template_enum(version_path, class_details, templates):
+    for template_info in templates:
+        class_file = Path(version_path, "resources",  "enum" + template_info["ext"])
+        if not os.path.exists(class_file):
+            if not (parent:=class_file.parent).exists():
+                parent.mkdir()
+            with open(class_file, "w", encoding="utf-8") as file:
+                header_file_path = os.path.join(
+                    os.getcwd(), "modernpython", "enum_header.py"
+                )
+                header_file = open(header_file_path, "r")
+                file.write(header_file.read())
+        with open(class_file, "a", encoding="utf-8") as file:
+            template_path = os.path.join(os.getcwd(), "modernpython/templates", template_info["filename"])
+            class_details["setInstances"] = _set_instances
+            with open(template_path, encoding="utf-8") as f:
+                args = {
+                    "data": class_details,
+                    "template": f,
+                    "partials_dict": partials,
+                }
+                output = chevron.render(**args)
+            file.write(output)
+
+def run_template_schema(version_path, class_details, templates):
```

Comment:
We both have a strong opinion on all resources in one file vs. 1 resource per file. 

And both are very valid use cases I believe. 

Maybe add an option to decide which one to choose?

Reply:
I don't have a strong opinion:) happy to have a look to make work cyclic reference with 1 resource per file, I didn't manage so far.